app/views/contexts/calculated_summary_context.py

```diff
@@ -162,18 +179,62 @@ def _get_formatted_total(self, groups, current_location, calculation_operator):
                     values_to_calculate.append(answer_value)
 
         calculated_total = calculation_operator(values_to_calculate)
+
+        return self._format_total(answer_format, calculated_total)
+
+    def _get_formatted_total_with_rule_evaluator(
+        self, groups, current_location, calculation
+    ):
+        answer_format = {"type": None}
+        for group in groups:
+            for block in group["blocks"]:
+                question = choose_question_to_display(
+                    block,
+                    self._schema,
+                    self._metadata,
+                    self._response_metadata,
+                    self._answer_store,
+                    self._list_store,
+                    current_location=current_location,
+                )
+                for answer in question["answers"]:
+                    if not answer_format["type"]:
+                        answer_format = {
+                            "type": answer["type"],
+                            "unit": answer.get("unit"),
+                            "unit_length": answer.get("unit_length"),
+                            "currency": answer.get("currency"),
+                        }
+
+        evaluate_calculated_summary = RuleEvaluator(
+            self._schema,
+            self._answer_store,
+            self._list_store,
+            self._metadata,
+            self._response_metadata,
+            location=current_location,
+        )
+
+        calculated_summary_total = evaluate_calculated_summary.evaluate(calculation)
+
+        return self._format_total(answer_format, calculated_summary_total)
+
+    @staticmethod
+    def _format_total(answer_format, total):
```

Comment:
Since you are adding all new methods we should type hint them? So type hint the whole file since there isn't much here?

Reply:
Added type hints to this module 👍

Reply:
Happy this is resolved.